purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -428,14 +503,73 @@ class Purchases internal constructor(
      * @param [packageToPurchase] The Package you wish to purchase
      * @param [listener] The listener that will be called when purchase completes.
      */
+    @Deprecated(
+        "Replaced with purchasePackageWithPurchaseOption",
+        ReplaceWith("purchasePackageWithPurchaseOption")
```

Comment:
Honestly, I have some concerns about just deprecating these methods... If people update the SDK but keep using these old methods, they won't get free trials automatically as they did until now. So that means a change in behavior but not a change in API, which will likely result in complains... not sure if we can should just remove it...

Reply:
maybe for these deprecated methods we SHOULD choose a free trial be default..

Reply:
Sounds about right - we have another ticket where we can deal with these, let's deal with them later: https://revenuecats.atlassian.net/browse/CF-907

Reply:
Well I have concerns about that approach, since it would be impossible to distinguish between a developer determined offer with a free trial and a "new customer acquisition" offer right? So we might be giving infinite free trials if a user adds a developer determined offer with a free trial to a product and uses this if I understand it correctly.

Reply:
That's why customers with multiple offers shouldn't be using that interface anyways... it should come with a big warning sign

Reply:
Well, we can put a warning in the docs but in the SDK we won't be able to differentiate between them to log any errors if they only have a single developer determined offer... This might be a calculated risk though. 

A way to lower that risk could be to try to check the purchase history and not give the user offers when using this deprecated function if they have purchased before or something like that... It would be adding a lot of complexity though and not sure if it's worth it.

Reply:
or we come back to maybe asking customers to put tags on developer-determined offers. and maybe we just always apply the non-dev-determined free trial, and then the next non-dev-determined offer 🤷‍♀️

Reply:
Right, I agree that it should work perfectly fine with the new API. My doubt is on whether we should keep the old API deprecated or just remove it. I'm just concerned some people wouldn't fully understand the changes and update to the latest SDK, keep using the deprecated API and start adding offers to some subscriptions. It's likely most people won't do this if they read the changelog but the danger would still exist. 

I think we need to decide between the convenience of keeping these deprecated methods vs the possible danger they bring. Probably worth talking about this this wednesday. Will keep the deprecated methods for now.